src/executables/csql_support.c

```diff
@@ -239,17 +239,97 @@ csql_invoke_system (const char *command)
     }
 }
 
+
+/*
+ * csql_invoke_formatter()
+ *   return: CSQL_SUCCESS/CSQL_FAILURE
+ *
+ * Note:
+ *   copy command editor buffer into temporary file and
+ *   invoke formatter. After the format is finished,
+ *   read the file into editor buffer.
+ */
+int
+csql_invoke_formatter ()
+{
+  auto[before_filename, before_fileptr] = filesys::open_temp_file ("bef_fmt_");
+  if (before_fileptr == NULL)
+    {
+      csql_Error_code = CSQL_ERR_OS_ERROR;
+      return CSQL_FAILURE;
+    }
+  filesys::auto_delete_file before_file_del (before_filename.c_str ());
+  filesys::auto_close_file before_file (before_fileptr);
+
+  if (csql_edit_write_file (before_file.get ()) == CSQL_FAILURE)
+    {
+      nonscr_display_error (csql_Scratch_text, SCRATCH_TEXT_LEN);
+      return CSQL_FAILURE;
+    }
+
+  auto[after_filename, after_fileptr] = filesys::open_temp_file ("aft_fmt_");
+  if (after_fileptr == NULL)
+    {
+      csql_Error_code = CSQL_ERR_OS_ERROR;
+      return CSQL_FAILURE;
+    }
+  filesys::auto_delete_file after_file_del (after_filename.c_str ());
+  filesys::auto_close_file after_file (after_fileptr);
+
+  char *cmd = csql_get_tmp_buf (strlen (csql_Formatter_cmd) + 1 + before_filename.size () + 1 + after_filename.size ());
+  if (cmd == NULL)
+    {
+      nonscr_display_error (csql_Scratch_text, SCRATCH_TEXT_LEN);
+      return CSQL_FAILURE;
+    }
+  fclose (after_file.release ());
+  sprintf (cmd, "%s %s > %s", csql_Formatter_cmd, before_filename.c_str (), after_filename.c_str ());
+
+  before_file.reset (fopen (before_filename.c_str (), "r"));
+  if (!before_file)
```

Comment:
which value **before_file** have?
if it is 0, what about exchange code like 

> if (system (cmd) != 0)

 (line:302)
 
 In my opinion, the code unity makes other reviewers and other users easy to figure out the code.

Reply:
It is the same as the csql_invoke_system_editor function.
it's like this
```
  file.reset (fopen (filename.c_str (), "r"));
  if (!file)
    {
      csql_Error_code = CSQL_ERR_OS_ERROR;
      nonscr_display_error (csql_Scratch_text, SCRATCH_TEXT_LEN);
      return CSQL_FAILURE;
    }
```

However, if the code is difficult to understand, I will also change the csql_invoke_system_editor function in regacy code.

Reply:
I think instead of changing legacy code, the adding comment will be better.

We cannot expect what error occur because this is legacy code.

Or if you think this is necessary to change, another issue should be handled this problem.

Reply:
Ok. I understand. I'll add comment.
Thank you.